tools/pyboard.py

```diff
@@ -287,11 +287,15 @@ def __init__(
             for attempt in range(wait + 1):
                 try:
                     if os.name == "nt":
-                        # Windows does not set DTR or RTS by default
                         self.serial = serial.Serial(**serial_kwargs)
-                        self.serial.dtr = True
-                        self.serial.rts = False
                         self.serial.port = device
+                        portinfo = list(serial.tools.list_ports.grep(device))  # type: ignore
+                        if portinfo and portinfo[0].manufacturer != "Microsoft":
+                            # ESPxx boards and ESP32 boards use RTS/CTS for flashing and boot mode selection
+                            # DTR False, to avoid using the reset button will hang the MCU in bootloader mode
+                            # RTS False, to prevent pulses on rts on serial.close() that would POWERON_RESET an ESPxx
+                            self.serial.dtr = False  # DTR False = gpio0 High = Normal boot
```

Comment:
I'm not sure why this was previously True and is now False?

Also, if Windows doesn't set DTR/RTS by default, wouldn't it be good to set them for all drivers?  Why can't it just set them both False for all drivers?  Or at least True/False for Microsoft drivers?

Reply:
I added the reasoning in the Issue , perhaps not the best place.
let me copy it here :

any unconditional combination ends up blocking normal operation on some type of board or another.
I suspect that the non-microsoft drivers may be less strict in the de-initialization , causing the spikes, but that is just speculation.

## A simple overview (for windows only) : 
DTR | CTS | ESP8266 |ESP32 | STM32 | SAMD51 | RP2040
--- | --- | --- | --- | --- | --- | --
unspecified| unspecified | ❌ Win32:Reset on disconnect |❌ Win32:Reset on disconnect |✅ OK|✅ OK|✅ OK
True| False | ❌ Win32:Hang in bootloader |❌ Win32:Hang in bootloader |✅ OK |✅ OK |✅ OK
False| False | ✅ OK |✅ OK|❌ No Repl|❌ No Repl| ❌ No Repl
True| True | ❌ Win32:Reset on disconnect | ❌ Win32:Reset on disconnect | ❌ No Repl  | ❌ No Repl | ❌ No Repl
False | True | ❌ Win32 : Reset on disconnect|❌ Win32 : Reset on disconnect| ❌ No Repl  | ❌ no Repl | ❌ No Repl
serial.manufacturer  | |wch.cn|Silicon Labs| Microsoft| Microsoft| Microsoft
 serial.description  | |USB-SERIAL CH340 |Silicon Labs CP210x USB to UART Bridge  | USB Serial Device | USB Serial Device | USB Serial Device 


### Updated Logic
The updated logic will only set the dtr/rts signals for boards that do not use standard Microsoft drivers( based on the manufacturer)
It would also be possible to check against a list of known driver manufactures (like `wch.cn` or `Silicon Labs` ) but this would require a list of known drivers for all ports. ( which I do not have) 

serial.manufacturer   | DTR | CTS | ESP8266 |ESP32 | STM32 | SAMD51 | RP2040
--- | --- | --- | --- | --- | --- | --- | --
== "Microsoft" | unspecified| unspecified | - | - |✅ OK|✅ OK|✅ OK
!= "Microsoft" (wch.cn,Silicon Labs)| False| False | ✅ OK |✅ OK| - | - | -

Reply:
OK, thanks for the details.  That makes sense then.